@hartwork this is interesting. Tests are green on Python 3.14 but fails on 3.10 (and also on any <3,14 on my local box). Seems this PR is OK, but our tests are not robust enough regarding asyncio timings which are possible a tiny bit different on different python versions.

This is the first failing test using uv with python 3.10
```
tami@tami-Precision-7560:~/pychess$ PYCHESS_UNITTEST=true PYTHONPATH=lib:testing uv run --python 3.10 python -m unittest testing.ficsplay.PlayGameTests.test3 -v
test3 (testing.ficsplay.PlayGameTests)
Accepting a seek ... 
(python3 -m unittest:490897): Gtk-WARNING **: 23:37:11.279: Can't set a parent on widget which has a parent
/home/tami/pychess/lib/pychess/perspectives/fics/__init__.py:79: DeprecationWarning: Gtk.ToolButton.new_from_stock is deprecated
  self.logoff_button = Gtk.ToolButton.new_from_stock(Gtk.STOCK_DISCONNECT)
/home/tami/pychess/lib/pychess/perspectives/fics/ParrentListSection.py:47: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/Projects/PyGObject/InitializerDeprecations
  menu_item = Gtk.MenuItem(label)
/home/tami/pychess/lib/pychess/widgets/ChannelsPanel.py:168: DeprecationWarning: Gtk.ScrolledWindow.add_with_viewport is deprecated
  self.add_with_viewport(vbox)
/home/tami/pychess/lib/pychess/widgets/ChannelsPanel.py:58: DeprecationWarning: Gdk.Cursor.new is deprecated
  self.stdcursor = Gdk.Cursor.new(Gdk.CursorType.LEFT_PTR)
/home/tami/pychess/lib/pychess/perspectives/fics/ChatPanel.py:44: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/Projects/PyGObject/InitializerDeprecations
  notebook.append_page(self.viewspanel, Gtk.Label(_("Chat")))
/home/tami/pychess/lib/pychess/perspectives/fics/ChatPanel.py:45: PyGTKDeprecationWarning: Using positional arguments with the GObject constructor has been deprecated. Please specify keyword(s) for "label" or use a class specific constructor. See: https://wiki.gnome.org/Projects/PyGObject/InitializerDeprecations
  notebook.append_page(self.infopanel, Gtk.Label(_("Info")))
/home/tami/pychess/lib/pychess/perspectives/fics/ConsolePanel.py:80: DeprecationWarning: Gtk.Widget.modify_font is deprecated
  self.readView.modify_font(fontdesc)

(python3 -m unittest:490897): Gtk-WARNING **: 23:37:11.503: Attempting to add a widget with type GtkEntry to a container of type GtkBox, but the widget is already inside a container of type pychess+widgets+ChatView+ChatView, please remove the widget from its existing container first.
/home/tami/pychess/lib/pychess/widgets/Background.py:239: DeprecationWarning: Gtk.StyleContext.get_background_color is deprecated
  color = widget.get_style_context().get_background_color(Gtk.StateFlags.NORMAL)
Task cancelled <Task cancelled name='Task-4' coro=<GameModel.start.<locals>.coro() done, defined at /home/tami/pychess/lib/pychess/Utils/GameModel.py:828> created at /home/tami/.local/share/uv/python/cpython-3.10.19-linux-x86_64-gnu/lib/python3.10/asyncio/tasks.py:337>
FAIL

======================================================================
FAIL: test3 (testing.ficsplay.PlayGameTests)
Accepting a seek
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tami/.local/share/uv/python/cpython-3.10.19-linux-x86_64-gnu/lib/python3.10/unittest/async_case.py", line 64, in _callTestMethod
    self._callMaybeAsync(method)
  File "/home/tami/.local/share/uv/python/cpython-3.10.19-linux-x86_64-gnu/lib/python3.10/unittest/async_case.py", line 87, in _callMaybeAsync
    return self._asyncioTestLoop.run_until_complete(fut)
  File "/home/tami/.local/share/uv/python/cpython-3.10.19-linux-x86_64-gnu/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/home/tami/.local/share/uv/python/cpython-3.10.19-linux-x86_64-gnu/lib/python3.10/unittest/async_case.py", line 101, in _asyncioLoopRunner
    ret = await awaitable
  File "/home/tami/pychess/testing/ficsplay.py", line 294, in test3
    self.assertEqual(gamemodel.ply, 3)
AssertionError: 1 != 3

----------------------------------------------------------------------
Ran 1 test in 10.305s

FAILED (failures=1)
```

--- a/lib/pychess/ic/managers/BoardManager.py
+++ b/lib/pychess/ic/managers/BoardManager.py
@@ -624,10 +624,14 @@ class BoardManager(GObject.GObject):
                 if game.relation == IC_POS_OBSERVING_EXAMINATION:
                     game.relation = relation  # IC_POS_EXAMINATING
                     # print("IC_POS_OBSERVING_EXAMINATION --> IC_POS_EXAMINATING")
-                    # before this change server sent an unobserve to us
-                    # and it removed gameno from started events dict
-                    # we have to put it back...
-                    self.gamemodelStartedEvents[game.gameno] = asyncio.Event()
+                    # The existing observed game model keeps running. FICS may
+                    # send an unobserve line first, which removes the started
+                    # marker, so restore it as already started instead of
+                    # blocking the following style12 moves forever.
+                    event = self.gamemodelStartedEvents.setdefault(
+                        game.gameno, asyncio.Event()
+                    )
+                    event.set()
                     return
 
             else:

--- a/lib/pychess/perspectives/games/__init__.py
+++ b/lib/pychess/perspectives/games/__init__.py
@@ -677,7 +677,7 @@ class Games(GObject.GObject, Perspective):
             leaf = self.dock.dock(
                 self.docks["board"][1],
                 CENTER,
-                Gtk.Label(label=self.docks["board"][0]),
+                self.docks["board"][0],
                 "board",
             )
             self.docks["board"][1].show_all()

--- a/testing/ficslecturebot.py
+++ b/testing/ficslecturebot.py
@@ -136,16 +136,20 @@ class ObserveGameTests(EmittingTestCase):
         await self.runAndAssertEquals(signal, lines, expectedResults)
 
         def on_gmwidg_created(persp, gmwidg, event):
-            event.set()
+            gamemodel = self.games_persp.cur_gmwidg().gamemodel
+
+            def on_game_started(game, event):
+                event.set()
+
+            gamemodel.connect("game_started", on_game_started, event)
+            self.gamemodel = gamemodel
 
         event = asyncio.Event()
         self.games_persp.connect("gmwidg_created", on_gmwidg_created, event)
 
         await asyncio.wait_for(event.wait(), timeout=5)
 
-        gamemodel = self.games_persp.cur_gmwidg().gamemodel
-
-        print(gamemodel)
+        print(self.gamemodel)
 
         lines = [
             # bsetup
@@ -212,14 +216,12 @@ class ObserveGameTests(EmittingTestCase):
 
         await self.connection.process_lines(lines)
 
-        self.assertEqual(game.move_queue.qsize(), 10)
-
         # let the game model process the moves
         await asyncio.sleep(0)
 
-        print(gamemodel.boards[-1])
+        print(self.gamemodel.boards[-1])
 
-        self.assertEqual(gamemodel.ply, 0)
+        self.assertEqual(self.gamemodel.ply, 3)
 
         lines = [
             # kibitz Example 2:...
@@ -277,11 +279,12 @@ class ObserveGameTests(EmittingTestCase):
 
         await self.connection.process_lines(lines)
 
-        self.assertEqual(game.move_queue.qsize(), 13)
+        # let the game model process the moves
+        await asyncio.sleep(0)
 
-        print(gamemodel.boards[-1])
+        print(self.gamemodel.boards[-1])
 
-        self.assertEqual(gamemodel.ply, 0)
+        self.assertEqual(self.gamemodel.ply, 1)
 
 
 if __name__ == "__main__":

--- a/testing/ficsplay.py
+++ b/testing/ficsplay.py
@@ -239,11 +239,18 @@ class PlayGameTests(EmittingTestCase):
         await self.runAndAssertEquals("playGameCreated", lines, (game,))
 
         def on_gmwidg_created(persp, gmwidg, event):
-            event.set()
+            gamemodel = self.games_persp.cur_gmwidg().gamemodel
+
+            def on_game_started(game, event):
+                p0, p1 = gamemodel.players
+                p1.move_queue.put_nowait(Move(newMove(G8, F6)))
+                event.set()
+
+            gamemodel.connect("game_started", on_game_started, event)
+            self.gamemodel = gamemodel
 
         event = asyncio.Event()
         self.games_persp.connect("gmwidg_created", on_gmwidg_created, event)
-
         await asyncio.wait_for(event.wait(), timeout=5)
 
         lines = [
@@ -257,6 +264,9 @@ class PlayGameTests(EmittingTestCase):
 
         self.assertEqual(game.move_queue.qsize(), 1)
 
+        # let the game model process the moves
+        await asyncio.sleep(0)
+
         lines = [
             BLOCK_START + "59" + BLOCK_SEPARATOR + "1" + BLOCK_SEPARATOR,
             "<12> rnbqkb-r pppppppp -----n-- -------- -------- -P------ P-PPPPPP RNBQKBNR W -1 1 1 1 1 1 85 WLTL gbtami -1 1 0 39 39 60000 60000 2 N/g8-f6 (0:00.000) Nf6 1 1 0",
@@ -268,29 +278,13 @@ class PlayGameTests(EmittingTestCase):
 
         await self.connection.process_lines(lines)
 
-        self.assertEqual(game.move_queue.qsize(), 3)
-
-        gamemodel = self.games_persp.cur_gmwidg().gamemodel
-
-        def on_game_started(game):
-            p0, p1 = gamemodel.players
-            p1.move_queue.put_nowait(Move(newMove(G8, F6)))
-
-            # My opp is an ICPlayer and his move_queue is gamemodel.ficsgame.move_queue
-            # BoardManager fed it with all the moves via onStyle12
-            assert isinstance(p0, ICPlayer)
-            self.assertEqual(p0.move_queue.qsize(), 3)
-
-            assert isinstance(p1, Human)
-            self.assertEqual(p1.move_queue.qsize(), 1)
-
-        gamemodel.connect("game_started", on_game_started)
+        self.assertEqual(game.move_queue.qsize(), 2)
 
         # let the game model process the moves
         await asyncio.sleep(0)
 
-        self.assertEqual(gamemodel.ply, 3)
-        print(gamemodel.boards[-1])
+        self.assertEqual(self.gamemodel.ply, 3)
+        print(self.gamemodel.boards[-1])
 
 
 if __name__ == "__main__":

--- a/testing/ficsplay.py
+++ b/testing/ficsplay.py
@@ -21,8 +21,6 @@ from pychess.ic.FICSObjects import (
 )
 from pychess.ic import BLOCK_START, BLOCK_SEPARATOR, BLOCK_END
 from ficsmanagers import EmittingTestCase
-from pychess.Players.Human import Human
-from pychess.Players.ICPlayer import ICPlayer
 from pychess.System.Log import log
 
 log.logger.setLevel(logging.DEBUG)

--- a/testing/ficspuzzlebot.py
+++ b/testing/ficspuzzlebot.py
@@ -51,6 +51,29 @@ class ExamineGameTests(EmittingTestCase):
     async def asyncTearDown(self):
         await cancel_all_tasks()
 
+    async def runAndWaitForGameStarted(self, signal, lines, expectedResults):
+        event = asyncio.Event()
+        result = {}
+
+        def on_gmwidg_created(persp, gmwidg):
+            gamemodel = gmwidg.gamemodel
+            result["gamemodel"] = gamemodel
+
+            def on_game_started(gamemodel):
+                event.set()
+
+            gamemodel.connect("game_started", on_game_started)
+
+        handler_id = self.games_persp.connect("gmwidg_created", on_gmwidg_created)
+        try:
+            await self.runAndAssertEquals(signal, lines, expectedResults)
+            await asyncio.wait_for(event.wait(), timeout=5)
+        finally:
+            if self.games_persp.handler_is_connected(handler_id):
+                self.games_persp.disconnect(handler_id)
+
+        return result["gamemodel"]
+
     async def test1(self):
         """Test puzzlebot starting a new mate in 2 puzzle"""
 
@@ -86,17 +109,7 @@ class ExamineGameTests(EmittingTestCase):
         )
         game = self.connection.games.get(game)
         expectedResults = (game,)
-        await self.runAndAssertEquals(signal, lines, expectedResults)
-
-        def on_gmwidg_created(persp, gmwidg, event):
-            event.set()
-
-        event = asyncio.Event()
-        self.games_persp.connect("gmwidg_created", on_gmwidg_created, event)
-
-        await asyncio.wait_for(event.wait(), timeout=5)
-
-        gamemodel = self.games_persp.cur_gmwidg().gamemodel
+        gamemodel = await self.runAndWaitForGameStarted(signal, lines, expectedResults)
         print(gamemodel)
 
         p0, p1 = gamemodel.players
@@ -292,17 +305,7 @@ class ExamineGameTests(EmittingTestCase):
         )
         game = self.connection.games.get(game)
         expectedResults = (game,)
-        await self.runAndAssertEquals(signal, lines, expectedResults)
-
-        def on_gmwidg_created(persp, gmwidg, event):
-            event.set()
-
-        event = asyncio.Event()
-        self.games_persp.connect("gmwidg_created", on_gmwidg_created, event)
-
-        await asyncio.wait_for(event.wait(), timeout=5)
-
-        gamemodel = self.games_persp.cur_gmwidg().gamemodel
+        gamemodel = await self.runAndWaitForGameStarted(signal, lines, expectedResults)
         print(gamemodel)
 
         lines = [

